sematic/scheduling/job_scheduler.py

```diff
@@ -96,8 +96,23 @@ def update_run_status(
     The second is the updated external jobs tuple.
     """
     external_jobs = tuple(external_jobs)
+    active_jobs_remain = any(job.is_active() for job in external_jobs)
     if future_state.is_terminal():
-        return future_state, external_jobs
+        if active_jobs_remain:
+            logger.warning(
+                "There are still active external jobs for run in "
+                "state %s (will refresh jobs): %s",
+                future_state,
+                external_jobs,
```

Comment:
Please add the run id in the log. You will have to add it to the function parameters as well.
```suggestion
            logger.warning(
                "There are still active external jobs for run %s in "
                "state %s (will refresh jobs): %s",
                run.id,
                future_state,
                external_jobs,
```

Reply:
the run id is part of the external job ids